src/widgets/settingspages/GeneralPage.cpp

```diff
@@ -455,7 +455,11 @@ void GeneralPage::initLayout(GeneralPageView &layout)
     layout.addDescription("All local files like settings and cache files are "
                           "store in this directory.");
     layout.addButton("Open AppData directory", [] {
+#ifdef Q_OS_DARWIN
+        QDesktopServices::openUrl("file://" + getPaths()->rootAppDataDirectory);
+#else
         QDesktopServices::openUrl(getPaths()->rootAppDataDirectory);
+#endif
```

Comment:
```suggestion
        QDesktopServices::openUrl("file://" + getPaths()->rootAppDataDirectory);
```

I think we could drop the `#ifdef` even, and always use the `file://` prefix on every OS (this is how it's done with the cache directory link.)
Tested on Linux with xdg-open/Nautilus, and seems to work fine. Would like additional tests on other OS and environments before going ahead with that though.

Reply:
I suspect this wouldn't work on Windows since it's not a valid path format ([see this](https://docs.microsoft.com/en-us/dotnet/standard/io/file-path-formats)). If that's the case I don't mind reversing this so `file://` is prepended on all OSes bar Windows. 

QT might handle `file://` though so I'll test that tomorrow.

Reply:
Thinking more like this if that's the case:

```c++
#ifdef Q_OS_WIN
        QDesktopServices::openUrl(getPaths()->rootAppDataDirectory);
#else
        QDesktopServices::openUrl("file://" + getPaths()->rootAppDataDirectory);
#endif

```

Reply:
Tested this morning on Windows 10 Enterprise. Prefixing `file://` to the string throws this output:
```
ShellExecute 'file://c/Users/USER/AppData/Roaming/Chatterino2' failed (error 5).
```

Since most Linux distros are fine with either do we want to make this a darwin-specific prefix or make it default with a different case for Windows? Either way it's still an ifdef.

Reply:
Thanks for testing it 👍

If we need an `#ifdef` either way, I think it's fine to keep the code as it is right now.

Reply:
Resolving this convo then, sweet as :+1: